src/types/index.ts

```diff
@@ -20,6 +21,11 @@ export interface L2Signer {
 
 export { Signer as L1Signer } from '@ethersproject/abstract-signer';
 
+export interface WalletConnection {
```

Comment:
not big fan of this `WalletConnection ` , can we make it more explicit ?like `WalletConnecionResponse`

Reply:
that's the type that holds l1 and l2Signer. it is not a kind of `response`